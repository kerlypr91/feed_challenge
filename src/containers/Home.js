import React, { Component } from 'react'
import { withRouter } from 'react-router'
import './Home.scss'

export class Home extends Component {
  handlePostsLinkClick = () => {
    const { history } = this.props

    history.push('/posts')
  }

  render () {
    return (
      <div className="home_container">
        <i className="fa fa-reddit-alien" />
        <span className="welcome">Welcome to the Feed Challenge</span>
        <span id="link" className="link" onClick={this.handlePostsLinkClick}><i className="fa fa-external-link" />Click here to see latest posts</span>
      </div>
    )
  }
}

export default withRouter(Home)