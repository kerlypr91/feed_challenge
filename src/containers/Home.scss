@import '../App.scss';

.home_container {
  color: white;
  display: flex;
  flex-direction: column;
  flex: 1;
  align-items: center;
  justify-content: center;

  > i {
    font-size: 100px;
  }

  .welcome {
    font-size: 70px;
    font-weight: bold;
    line-height: 80px;
    margin: 10px 0;
    text-align: center;
  }

  .link {
    margin: 10px;
    font-weight: bold;
    color: #e6e7e7;
    font-size: 18px;
    cursor: pointer;

    > i {
      margin-right: 10px;
    }

    @include app_link(white);
  }
}
