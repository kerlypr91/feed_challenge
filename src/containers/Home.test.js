import React from 'react';
import { shallow } from 'enzyme';
import { Home } from './Home';

describe('Home Tests', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Home />)

    expect(wrapper).toMatchSnapshot()
  });

  it('link should navigate to posts route', () => {
    const history = {
      push: jest.fn()
    }

    const wrapper = shallow(<Home history={history} />)
    const link = wrapper.find('#link')

    link.simulate('click')

    expect.assertions(3)
    expect(link).toBeTruthy()
    expect(history.push).toHaveBeenCalled()
    expect(history.push).toHaveBeenCalledWith('/posts')
  });
});