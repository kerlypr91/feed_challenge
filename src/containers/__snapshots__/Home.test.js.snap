// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Tests should match snapshot 1`] = `
<div
  className="home_container"
>
  <i
    className="fa fa-reddit-alien"
  />
  <span
    className="welcome"
  >
    Welcome to the Feed Challenge
  </span>
  <span
    className="link"
    id="link"
    onClick={[Function]}
  >
    <i
      className="fa fa-external-link"
    />
    Click here to see latest posts
  </span>
</div>
`;
